src/Microsoft.AspNet.OData.Shared/DeltaDeletedEntityObjectOfT.cs

```diff
@@ -0,0 +1,144 @@
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, dynamicDictionaryPropertyInfo: null, instanceAnnotationsPropertyInfo)
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>        
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, updatableProperties: null , dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+ 
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties"> Properties that can be updated</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, IEnumerable<string> updatableProperties, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : base(structuralType, updatableProperties, dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+            DeltaKind = EdmDeltaEntityKind.DeletedEntry;
+        }
+
+     
+        /// <inheritdoc />
+        public Uri Id
+        {
+            get
+            {
+                return _id;
+            }
+            set
+            {
+                _id = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public DeltaDeletedEntryReason Reason
```

Comment:
>DeltaDeletedEntryReason [](start = 15, length = 23)

Is it a nullable? 

public DeltaDeletedEntryReason? Reason {get;set;}

Reply:
Doesnt looked so

---
In reply to: [604612233](https://github.com/OData/WebApi/pull/2455#discussion_r604612233) [](ancestors = 604612233)

Reply:
Reason is optional -- we should be able to omit it from the response.  See [http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_DeletedEntity](http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_DeletedEntity)

---
In reply to: [604692550](https://github.com/OData/WebApi/pull/2455#discussion_r604692550) [](ancestors = 604692550,604612233)